podpac/core/data/datasource.py

```diff
@@ -444,17 +391,22 @@ def eval(self, coordinates, output=None):
             self._requested_source_coordinates_index = tuple(new_rsci)
 
         # get data from data source
-        self._requested_source_data = self._get_data()
+        self._requested_source_data = self._get_data(
+            self._requested_source_coordinates, self._requested_source_coordinates_index
+        )
 
         # if not provided, create output using the evaluated coordinates, or
         # if provided, set the order of coordinates to match the output dims
         # Note that at this point the coordinates are in the same CRS as the coordinates
-        if output is None:
+        if isinstance(self._requested_source_data, UnitsDataArray):
```

Comment:
It looks like the first case here is tested, but not the `elif output is None` and `else` clauses. This might be an appropriate time to add tests for these?

Reply:
Great job looking at the coverage here. It turns out those branches will NEVER be visited... the `_get_data` function already handles non`UnitDataArray` objects, so here we just have to deal with output existing, or not.